gateway/src/metrics.rs

```diff
@@ -279,34 +285,55 @@ impl Drop for Metrics {
 }
 
 impl MetricsConfig {
-    fn oximeter_collection_interval(&self) -> Duration {
-        Duration::from_secs(self.oximeter_collection_interval_secs as u64)
-    }
+    fn validate(self) -> anyhow::Result<ValidatedMetricsConfig> {
+        anyhow::ensure!(
+            self.oximeter_collection_interval_secs > 0,
+            "`metrics.oximeter_collection_interval_secs` probably shouldn't \
+             be 0 seconds",
+        );
+        let oximeter_collection_interval =
+            Duration::from_secs(self.oximeter_collection_interval_secs as u64);
+
+        anyhow::ensure!(
+            self.sp_poll_interval_ms > 0,
+            "`metrics.sp_poll_interval_ms` probably shouldn't be 0 ms",
+        );
+        let sp_poll_interval =
+            Duration::from_secs(self.sp_poll_interval_ms as u64);
```

Comment:
Should this either be `from_millis` or `sp_poll_interval_secs`?

Reply:
Whoops, good catch!